src/main/webapp/scripts/script.js

```diff
@@ -1,18 +1,23 @@
-const API_KEY = 'AIzaSyAr7MKPZKUCQ9G-N00I44EJqvsYjE3jAqU';
+const API_KEY = '';
 const YOUTUBE_DISCOVERY = 'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest';
 
-function loadApi(callback) {
-  gapi.load('client', function() {
-    gapi.client.init({
-      'apiKey': API_KEY,
-      'discoveryDocs': [YOUTUBE_DISCOVERY]
-    })
-    .then(function() {
-      callback();
-    })
-    .catch((error) => {
-      updateDom('Oops! An error occured with Youtube, please try again later.', 'popular-list-container');
-      console.log('Was not able to load Google API client:', error);
+/**
+ * Loads any Google APIs listed in the discoveryDocs
+ * @return {Promise} for completing the load
+ */
+function loadApi() {
+  return new Promise((resolve, reject) => {
+    gapi.load('client', () => {
+      gapi.client.init({
+        'apiKey': API_KEY,
+        'discoveryDocs': [YOUTUBE_DISCOVERY]
+      })
+      .then(resolve)
```

Comment:
Nit: Does this actually resolve the the promise? I thought the `then` shortcut just returns the single variable you have in it. So I'm thinking its not invoking promise `promise()` and just returns `promise` callback. Let me know if I'm wrong.

Reply:
this was part of PR#16, FuzzyTree advised me to change to this because it's equivalent to `.then(function() { resolve() })` so you're right its not invoking anything its just setting the callback function to resolve() which is what I want